src/sentry/static/sentry/app/index.js

```diff
@@ -1,153 +1,15 @@
-    ReactDOM.render(<Component />, rootEl);
-  } catch (err) {
-    // eslint-disable-next-line no-console
-    console.error(
-      new Error(
-        'An unencoded "%" has appeared, it is super effective! (See https://github.com/ReactTraining/history/issues/505)'
-      )
-    );
-    if (err.message === 'URI malformed') {
-      window.location.assign(window.location.pathname);
-    }
-  }
-};
-
-// The password strength component is very heavyweight as it includes the
-// zxcvbn, a relatively byte-heavy password strength estimation library. Load
-// it on demand.
-async function loadPasswordStrength(callback) {
-  const module = await import(/* webpackChunkName: "passwordStrength" */ 'app/components/passwordStrength');
-  callback(module);
+  window.csrfCookieName = csrfCookieName;
+  window.__sentryGlobalStaticPrefix = distPrefix;
+  window.__SENTRY__OPTIONS = sentryConfig;
+  window.__SENTRY__USER = userIdentity;
 }
 
-const globals = {
-  // This is the primary entrypoint for rendering the sentry app.
-  SentryRenderApp: () => render(Main),
-
-  // The following globals are used in sentry-plugins webpack externals
-  // configuration.
-  PropTypes,
-  React,
-  Reflux,
-  Router,
-  Sentry,
-  moment,
-  ReactDOM: {
-    findDOMNode: ReactDOM.findDOMNode,
-    render: ReactDOM.render,
-  },
-
-  // jQuery is still exported to the window as some bootsrap functionality
-  // makes use of it.
-  $: jQuery,
-  jQuery,
-
-  // django templates make use of these globals
-  createReactClass,
-};
-
-// The SentryApp global contains exported app modules for use in javascript
-// modules that are not compiled with the sentry bundle.
-globals.SentryApp = {
-  // The following components are used in sentry-plugins.
-  Form: require('app/components/forms/form').default,
-  FormState: require('app/components/forms/index').FormState,
-  LoadingIndicator: require('app/components/loadingIndicator').default,
-  plugins: {
-    add: plugins.add,
-    addContext: plugins.addContext,
-    BasePlugin: plugins.BasePlugin,
-    DefaultIssuePlugin: plugins.DefaultIssuePlugin,
-  },
-
-  // The following components are used in legacy django HTML views
-  passwordStrength: {load: loadPasswordStrength},
-  U2fSign: require('app/components/u2f/u2fsign').default,
-  ConfigStore: require('app/stores/configStore').default,
-  Alerts: require('app/components/alerts').default,
-  Indicators: require('app/components/indicators').default,
-  ProjectSelector: require('app/components/projectHeader/projectSelector').default,
-  Sidebar: require('app/components/sidebar').default,
-  SetupWizard: require('app/components/setupWizard').default,
-  OrganizationsLoader: require('app/components/organizations/organizationsLoader')
-    .default,
-};
-
-// Make globals available on the window object
-Object.keys(globals).forEach(name => (window[name] = globals[name]));
-
-export default globals;
+// Once data hydration is done we can initialize the app
+require('app/bootstrap');
```

Comment:
Wouldn't it be better if we wrapped all the init code in `app/bootstrap` above in a function and called it here instead giving us clearer module imports and more controlled init in tests?

This should also allow us to pass initial data as an argument to that function and allow us to stop relying on the `window` object for init in the fully separate FE scenario.

Reply:
Unfortunately the window.__* stuff is used in some other modules that are imported within the bootstrap.jsx. So it's a little more complicated, but yes I generally do agree.